config/src/main/java/org/opennms/integration/api/xml/schema/poller/DowntimeXml.java

```diff
@@ -0,0 +1,226 @@
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.integration.api.xml.schema.poller;
+
+import java.io.Serializable;
+import java.util.List;
+
+import javax.xml.bind.annotation.XmlAccessType;
+import javax.xml.bind.annotation.XmlAccessorType;
+import javax.xml.bind.annotation.XmlAttribute;
+import javax.xml.bind.annotation.XmlRootElement;
+import javax.xml.bind.annotation.XmlTransient;
+
+import com.google.common.collect.ImmutableList;
+
+/**
+ * Downtime model. This determines the rates at which addresses are to be
+ * polled when they remain down for extended periods. Usually polling is done
+ * at lower rates when a node is down until a certain amount of downtime at
+ * which the node is marked 'deleted'.
+ */
+
+@XmlRootElement(name="downtime")
+@XmlAccessorType(XmlAccessType.NONE)
+public class DowntimeXml implements Serializable {
+    private static final long serialVersionUID = 1634464860178402675L;
+
+    public static final String DELETE_ALWAYS = "always";
+    public static final String DELETE_MANAGED = "managed";
+    public static final String DELETE_NEVER = "never";
+
+    private static final List<String> s_deleteValues = ImmutableList.of(DELETE_ALWAYS, DELETE_MANAGED, DELETE_NEVER);
+
+    /**
+     * Start of the interval.
+     */
+    @XmlAttribute(name="begin")
+    private Long m_begin;
+
+    /**
+     * End of the interval.
+     */
+    @XmlAttribute(name="end")
+    private Long m_end;
+
+    /**
+     * Attribute that determines if service is to be deleted when down
+     * continuously since the start time.
+     */
+    private String m_delete;
+
+    /**
+     * Interval at which service is to be polled between the specified start
+     * and end when service has been continuously down.
+     */
+    @XmlAttribute(name="interval")
+    private Long m_interval;
+
+
+    public DowntimeXml() {
+        super();
+    }
+
+    public DowntimeXml(final long interval, final long begin, final long end) {
+        this();
+        setInterval(interval);
+        setBegin(begin);
+        setEnd(end);
+    }
+
+    public DowntimeXml(final long begin, final String delete) {
+        this();
+        setBegin(begin);
+        setDelete(delete);
+    }
+
+    /**
+     * Start of the interval.
+     */
+    public Long getBegin() {
+        return m_begin == null? 0 : m_begin;
```

Comment:
This is strange: It could either return a `long` or do not care about `null` values.